core/src/test/java/GameObjects/UrlBuilderImpTest.java

```diff
@@ -0,0 +1,95 @@
+package GameObjects;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for the UrlBuilder. The URLs provided were taken directly from the database using their
+ * API.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class UrlBuilderImpTest {
+  @Mock
+  GameSettings settingsMock;
+
+  @InjectMocks
+  UrlBuilderImp urlBuilder;
+
+  @Test
+  /**
+   * Test when general category is used with 9 questions and easy difficulty.
+   */
+  public void longGeneralEasy(){
```

Comment:
I'd prefer test method names of the form \<methodName\>\_\<state\>() for consistency with the file management tests.  E.g. "buildUrl_longGeneralEasy()".  However, the style guide suggests test\<methodName\>_\<state\>() (probably to avoid confusion with test helper methods), so we might change it in the future.